Remove alloca usage [RDY]

aravindprasads opened the pull request:
Fix for issue - https://github.com/radcli/radcli/issues/42 

Removed use of alloca as it is non-POSIX.

aravindprasads committed:
Changes to remove alloca

aravindprasads commented:
Hi @nmav ,

1) Can we remove alloca.h file too ? 

2) If this change looks fine, can we plan to merge it after the dictionary changes merge since the location of changes for alloca conflict with dictionary changes.

nmav commented on lib/dict.c:
```diff
@@ -42,7 +42,8 @@ int rc_read_dictionary (rc_handle *rh, char const *filename)
 	char            typestr[AUTH_ID_LEN];
 	char            optstr[AUTH_ID_LEN];
-	char            *cp, *ifilename;
+	char            ifilename[AUTH_ID_LEN] = {0};
```
not introduced in this patch but `AUTH_ID_LEN` has nothing to do with limits of filenames and is in fact very small (64). What about fixing it using `PATH_MAX`?

nmav commented on lib/dict.c:
```diff
@@ -265,12 +266,11 @@ int rc_read_dictionary (rc_handle *rh, char const *filename)
 				if (cp != NULL) {
-					ifilename = alloca(AUTH_ID_LEN);
 					*cp = '\0';
 					snprintf(ifilename, AUTH_ID_LEN, "%s/%s", filename, namestr);
```
we can use `sizeof(ifilename)` as you do above, to avoid copying around the size.

aravindprasads commented:
Hi @nmav 

Yes, changed size to PATH_MAX and fixed sizeof(). 

Kindly review and let me know your views. 

If the changes look fine, can we plan to merge it after the dictionary changes merge since the location of changes for alloca conflict with dictionary changes.

aravindprasads committed:
Merge pull request #2 from radcli/master

Merged to Master

aravindprasads committed:
Merge remote-tracking branch 'origin' into remove-alloca

aravindprasads commented:
Hi @nmav 

I have rebased the pull request with latest of Master branch. 

Kindly review the same.

nmav commented:
Thank you!
